cmd/commands/service/command.go

```diff
@@ -154,19 +156,46 @@ func (sc *serviceCommand) unlockIdentity(id, passphrase string) string {
 	}
 }
 
+func (sc *serviceCommand) tryRememberTOS(ctx *cli.Context) {
+	if !ctx.Bool(config.FlagAgreedTermsConditions.Name) {
+		return
+	}
+
+	go func() {
+		t := true
+		for i := 0; i < 5; i++ {
+			if err := sc.tequilapi.UpdateTerms(contract.TermsRequest{
+				AgreedProvider: &t,
+				AgreedVersion:  metadata.CurrentTermsVersion,
+			}); err == nil {
+				return
+			}
+			time.Sleep(time.Second * 2)
```

Comment:
So this stops the world for 10 seconds if I try to exit a node that is having issues with tequila? That's bad. Add something that's cancellable in case of an exit.

Reply:
No one really tracks this go routine, so nothing is really stopped. But you have fair point.